client/Components/login.jsx

```diff
@@ -31,10 +31,31 @@ const Login = (props) => {
     const user = token ? decodeToken(token) : null;
     if (user) {
       const userData = { username: user.username, id: user.id };
-      user && props.setUserValid(userData);
+      props.setUserValid(userData);
+      getUserData(userData);
     }
   }, []);
 
+  const getUserData = (data) => {
+    const userId = data.id;
+    fetch(`/api/profile/${userId}`)
+      .then((res) => {
+        if (!res.ok) {
+          props.setUserValid(res);
+          throw new Error("something went wrong fetching profile data");
+        }
+        return res.json();
+      })
+      .then((res) => {
+        if (res.username) {
+          props.setUserValid(res);
+        } else {
+          props.setUserValid(data);
+        }
+      })
+      .catch((err) => console.error(err));
```

Comment:
I believe once you change this up you can simplify things and not take a `data` parameter nor use any conditional logic on lines 50-52